src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResult.cs

```diff
@@ -69,62 +76,42 @@ public ImmutableArray<Diagnostic> GetAllDiagnostics(DiagnosticAnalyzer analyzer)
-        }
-
-        internal ImmutableArray<Diagnostic> GetSyntaxDiagnostics(ImmutableArray<DiagnosticAnalyzer> analyzers)
-        {
-            return GetDiagnostics(analyzers, getLocalSyntaxDiagnostics: true, getLocalSemanticDiagnostics: false, getNonLocalDiagnostics: false);
-        }
-
-        internal ImmutableArray<Diagnostic> GetSemanticDiagnostics(ImmutableArray<DiagnosticAnalyzer> analyzers)
-        {
-            return GetDiagnostics(analyzers, getLocalSyntaxDiagnostics: false, getLocalSemanticDiagnostics: true, getNonLocalDiagnostics: false);
+            return GetDiagnostics(Analyzers);
         }
 
-        internal ImmutableArray<Diagnostic> GetDiagnostics(IEnumerable<DiagnosticAnalyzer> analyzers, bool getLocalSyntaxDiagnostics, bool getLocalSemanticDiagnostics, bool getNonLocalDiagnostics)
+        private ImmutableArray<Diagnostic> GetDiagnostics(IEnumerable<DiagnosticAnalyzer> analyzers)
         {
             var excludedAnalyzers = Analyzers.Except(analyzers);
             var excludedAnalyzersSet = excludedAnalyzers.Any() ? excludedAnalyzers.ToImmutableHashSet() : ImmutableHashSet<DiagnosticAnalyzer>.Empty;
-            return GetDiagnostics(excludedAnalyzersSet, getLocalSyntaxDiagnostics, getLocalSemanticDiagnostics, getNonLocalDiagnostics);
+            return GetDiagnostics(excludedAnalyzersSet);
         }
 
-        private ImmutableArray<Diagnostic> GetDiagnostics(ImmutableHashSet<DiagnosticAnalyzer> excludedAnalyzers, bool getLocalSyntaxDiagnostics, bool getLocalSemanticDiagnostics, bool getNonLocalDiagnostics)
+        private ImmutableArray<Diagnostic> GetDiagnostics(ImmutableHashSet<DiagnosticAnalyzer> excludedAnalyzers)
         {
-            if (SyntaxDiagnostics.Count > 0 || SemanticDiagnostics.Count > 0 || CompilationDiagnostics.Count > 0)
+            if (SyntaxDiagnostics.Count > 0 || SemanticDiagnostics.Count > 0 || AdditionalFileDiagnostics.Count > 0 || CompilationDiagnostics.Count > 0)
             {
                 var builder = ImmutableArray.CreateBuilder<Diagnostic>();
-                if (getLocalSyntaxDiagnostics)
-                {
-                    AddLocalDiagnostics(SyntaxDiagnostics, excludedAnalyzers, builder);
-                }
-
-                if (getLocalSemanticDiagnostics)
-                {
-                    AddLocalDiagnostics(SemanticDiagnostics, excludedAnalyzers, builder);
-                }
-
-                if (getNonLocalDiagnostics)
-                {
-                    AddNonLocalDiagnostics(CompilationDiagnostics, excludedAnalyzers, builder);
-                }
+                AddLocalDiagnostics(SyntaxDiagnostics, excludedAnalyzers, builder);
+                AddLocalDiagnostics(SemanticDiagnostics, excludedAnalyzers, builder);
+                AddLocalDiagnostics(AdditionalFileDiagnostics, excludedAnalyzers, builder);
+                AddNonLocalDiagnostics(CompilationDiagnostics, excludedAnalyzers, builder);
```

Comment:
What is the distinction between local and nonlocal diagnostics?

Reply:
Non-local diagnostics are basically compilation end diagnostics and no location diagnostics - basically those cannot be produced by analyzing a single file or do not have a location in source.